### ISSUE TITLE
Proposal: Yarn 3 and PnP support

### ISSUE THREAD

**Issue by daniel-brenot** [eclipse-theia/theia#11924]
### Feature Description:
While this doesn't necessarily mean that the project needs to use yarn 3, having support for it would fix quite a few issues. For one, the project currently relies on links to files that exists within the node modules folder. Due to this, the only package resolution algorithm that can be used is one where every workspace has a copy of all the node_modules needed for the package, rather than having a single folder with the files, or even just resolving using pnp with no node modules.

If yarn 3 is used, the application will install much quicker. At the moment there may be a blocking point of node-pty due to its use of node-gyp, but I am currently working on that issue (https://github.com/microsoft/node-pty/issues/507). 

Let me know if this is of any interest. I am currently working on support and hope to commit it upstream if all goes well, but i wanted to make sure that the work would get merged.

**Comment by nikitinas** [comment]
@daniel-brenot Please, do that! I have terrible pain with yarn classic due to my corporate proxy. Yarn 3 works fine, but I need Theia to support it. If you need any help with yarn 3 migration, please, contact me.

**Comment by danepowell** [comment]
This will become more urgent with issues like https://github.com/isaacs/jackspeak/issues/5

tl;dr Yarn 1 has bugs in its version resolution logic that make it impossible to use with certain packages using more advanced version definitions. The Yarn maintainer has mentioned that this will become more commonplace in the future, and we really need to migrate to a newer version of Yarn.
